lib/bulk_processor/s3_file.rb

```diff
@@ -38,9 +38,18 @@ def open
     # @param contents [String] the contents of the file to create
     # @return [String] the URL of the new file
     def write(contents)
-      remote_file = resource.bucket(bucket).object(key)
-      remote_file.put(body: contents)
-      remote_file.public_url
+      retry_limit = 3
+      begin
+        remote_file = resource.bucket(bucket).object(key)
+        remote_file.put(body: contents)
```

Comment:
have you tested that `remote_file.put` will throw an error? I thought the issue we were having was this did not throw an error when it failed

Reply:
yes I tested in console, it gave me this error 
`Traceback (most recent call last): 1: from (irb):13 Aws::S3::Errors::AccessDenied (Access Denied)`

Reply:
I don't know much about the specific issue, but I would expect access denied error to not be resolved by retries (since we either have permissions to access the file or not).

Reply:
looks like permission issues are gone if we try to connect to production bucket.